### PR TITLE
feat: swap revamp

### PR DESCRIPTION
**Pull request by Chef-Yogi**
---

## PR-Codex overview
This PR primarily focuses on enhancing the `Swap` functionality within the PancakeSwap application, introducing new components, improving existing ones, and refining styles and configurations for better user experience and performance.

### Detailed summary
- Added `SwapUIV2` export in `packages/widgets-internal/swap-v2/index.ts`.
- Introduced `SwapLoading` component for loading states.
- Updated `GAS_PRICE` enum values for dynamic pricing.
- Improved layout and styling in various components, including `SwapChart` and `Modal`.
- Added `MotionTabMenu` and related props for better tab navigation.
- Enhanced `CurrencyLogo` with image reference support.
- Updated `ConnectWalletButton` to include an icon option.
- Integrated new hooks for managing user balances and price impacts.
- Refined routing and configuration for swap-related pages.

> The following files were skipped due to too many changes: `packages/uikit/src/components/Svg/Icons/WalletFilledV2.tsx`, `packages/widgets-internal/swap-v2/NumericalInput.tsx`, `packages/widgets-internal/swap-v2/CurrencyInputPanelSimplify.tsx`, `apps/web/src/components/Svg/RefreshIcon.tsx`, `packages/uikit/src/components/CopyButton/CopyButton.tsx`, `packages/widgets-internal/swap-v2/Collapse.tsx`, `apps/web/src/components/Menu/GlobalSettings/SettingsModalV2/ExpertModeTab.tsx`, `packages/uikit/src/components/MotionTabMenu/MotionTabMenu.tsx`, `apps/web/src/components/AddToWallet/AddToWalletButton.tsx`, `apps/web/src/views/Swap/V3Swap/components/SlippageButton.tsx`, `apps/web/src/views/SwapSimplify/V4Swap/FlipButton.tsx`, `packages/uikit/src/components/Svg/Icons/Verified`, `apps/web/src/views/SwapSimplify/V4Swap/TradeDetails.tsx`, `apps/web/src/components/Menu/GlobalSettings/GasSettings.tsx`, `apps/web/src/components/Menu/GlobalSettings/SettingsModalV2/SettingsTab.tsx`, `apps/web/src/components/SearchModal/CommonBases.tsx`, `packages/uikit/src/widgets/Modal/MotionModal.tsx`, `apps/web/src/components/SearchModal/CurrencySearchModal.tsx`, `apps/web/src/views/SwapSimplify/V4Swap/RoutesBreakdown.tsx`, `apps/web/src/views/SwapSimplify/V4Swap/SwapSelectionTab.tsx`, `apps/web/src/views/Swap/components/SwapModalHeaderV2.tsx`, `apps/web/src/views/SwapSimplify/index.tsx`, `packages/uikit/src/components/Skeleton/Skeleton.tsx`, `packages/localization/src/config/translations.json`, `apps/web/src/views/SwapSimplify/V4Swap/AddressInputPanel.tsx`, `apps/web/src/views/Swap/V3Swap/components/TransactionConfirmSwapContentV2.tsx`, `apps/web/src/components/Menu/GlobalSettings/SettingsModalV2/CustomizeRoutingTab.tsx`, `apps/web/src/components/Menu/GlobalSettings/SettingsModalV2/index.tsx`, `apps/web/src/views/SwapSimplify/V4Swap/FormMainV4.tsx`, `apps/web/src/components/Menu/GlobalSettings/TransactionSettings.tsx`, `apps/web/src/views/SwapSimplify/V4Swap/index.tsx`, `apps/web/src/views/SwapSimplify/V4Swap/AdvancedSwapDetails.tsx`, `apps/web/src/views/Swap/V3Swap/components/SwapModalFooterV2.tsx`, `apps/web/src/views/Swap/V3Swap/containers/ConfirmSwapModalV2.tsx`, `apps/web/src/components/CurrencyInputPanelSimplify/index.tsx`, `apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx`, `apps/web/src/views/SwapSimplify/V4Swap/SwapCommitButton.tsx`, `apps/web/public/images/swap/Arrow_Lottie.json`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: packages/widgets-internal/swap-v2/CurrencyInputPanelSimplify.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { NumericalInput, NumericalInputProps } from "./NumericalInput";
+
+interface CurrencyInputPanelProps extends Omit<NumericalInputProps, "onBlur" | "OnFocus"> {
+  onInputBlur?: () => void;
+  onInputFocus?: () => void;
+  id: string;
+  top?: React.ReactNode;
+  bottom?: React.ReactNode;
+  inputLeft?: React.ReactNode;
+  showBridgeWarning?: boolean;
```
Comment (chefjackson) [inline packages/widgets-internal/swap-v2/CurrencyInputPanelSimplify.tsx:12]: what's this property for?

Comment (Chef-Yogi) [reply packages/widgets-internal/swap-v2/CurrencyInputPanelSimplify.tsx:12]: It seems related to the current Aptos swap input panel; if we don't need to apply this swap-revamp to Aptos in the future, then we don't need this for now

File: apps/web/next.config.mjs [hunk 2]
```diff
@@ -225,6 +225,23 @@ const config = {
...
+            // TODO: remove after revamp is live
+            source: '/revamp-swap',
+            destination: '/swap',
+            permanent: false,
+          }
+        : {
+            source: '/swap-revamp',
+            destination: '/revamp-swap',
+            permanent: false,
+          },
```
Comment (chefjackson) [inline apps/web/next.config.mjs:239]: shall we remove?

File: apps/web/next.config.mjs [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable @typescript-eslint/no-var-requires */
 import BundleAnalyzer from '@next/bundle-analyzer'
 import { withWebSecurityHeaders } from '@pancakeswap/next-config/withWebSecurityHeaders'
-import smartRouterPkgs from '@pancakeswap/smart-router/package.json' with { type: 'json' }
+import smartRouterPkgs from '@pancakeswap/smart-router/package.json' assert { type: 'json' }
```
Comment (chefjackson) [inline apps/web/next.config.mjs:4]: any reason for the update here?

File: apps/web/src/pages/revamp-swap/index.tsx
Comment (chefjackson) [inline apps/web/src/pages/revamp-swap/index.tsx:1]: we don't need this page anymore

File: apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+    <FlexGap alignItems="flex-start">
+      <FlexGap justifyContent="center" alignItems="flex-start" flexDirection="column" gap="8px">
+        <Text>
+          {t(
+            'Final execution price may be differ from the market price due to trader size, available liquidity, and trading route. Please proceed with caution.',
+          )}
+        </Text>
+        {/* <StyledLinkExternal color="primary60" external href="https://www.hashdit.io" showExternalIcon={false}>
+          {t('Learn More')}
+        </StyledLinkExternal> */}
```
Comment (ChefMomota) [inline apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx:202]: Remove it, if not use

File: apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+          {t(
+            'You may only get the amount of “Minimum received” with a high slippage setting. Reset your slippage to avoid potential losses.',
+          )}
+        </Text>
+        <StyledTextButton onClick={() => setUserSlippageTolerance(DEFAULT_SLIPPAGE_TOLERANCE)} scale="sm" px="0">
+          {t('Reset Slippage')}
+        </StyledTextButton>
+        {/* <StyledLinkExternal color="primary60" external href="https://www.hashdit.io" showExternalIcon={false}>
+          {t('Learn More')}
+        </StyledLinkExternal> */}
```
Comment (ChefMomota) [inline apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx:225]: remove

File: apps/web/src/state/types.ts [hunk 6]
```diff
@@ -14,9 +14,9 @@ import { Address, parseEther } from 'viem'
 import { NftToken } from './nftMarket/types'
 
 export enum GAS_PRICE {
-  default = '1',
-  fast = '4',
-  instant = '5',
+  default = '5',
```
Comment (chefjackson) [inline apps/web/src/state/types.ts:17]: any context on this update?

File: apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+    }
+    if (isRiskToken1) {
+      count++
+    }
+    if (isPriceImpactTooHigh) {
+      count++
+    }
+    if (isSlippageTooHigh) {
+      count++
+    }
```
Comment (ChefMomota) [inline apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx:299]: Can we change it 

if (isRiskToken0 || isRiskToken1 || isPriceImpactTooHigh || isSlippageTooHigh) {
   count++
}

File: apps/web/src/components/CurrencyInputPanelSimplify/index.tsx [hunk 8]
```diff
@@ -0,0 +1,371 @@
...
+  title,
+  isUserInsufficientBalance,
+}: CurrencyInputPanelProps) {
+  const { address: account } = useAccount()
+  const selectedCurrencyBalance = useCurrencyBalance(account ?? undefined, currency ?? undefined)
+  const { t } = useTranslation()
+
+  const mode = id
+  const token = pair ? pair.liquidityToken : currency?.isToken ? currency : null
+  // const tokenAddress = token ? safeGetAddress(token.address) : null
```
Comment (ChefMomota) [inline apps/web/src/components/CurrencyInputPanelSimplify/index.tsx:212]: remove

File: apps/web/next.config.mjs (hunk 3 above)
Comment (Chef-Yogi) [reply apps/web/next.config.mjs:4]: @thechefpenguin, could you take a look at this?

File: apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx (hunk 5 above)
Comment (Chef-Yogi) [reply apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx]: Done

File: apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx (hunk 4 above)
Comment (Chef-Yogi) [reply apps/web/src/components/AccessRisk/SwapRevampRiskDisplay.tsx]: Done

File: apps/web/src/pages/revamp-swap/index.tsx
Comment (Chef-Yogi) [reply apps/web/src/pages/revamp-swap/index.tsx:1]: done

File: apps/web/next.config.mjs (hunk 2 above)
Comment (Chef-Yogi) [reply apps/web/next.config.mjs]: done, removed

File: apps/web/next.config.mjs (hunk 3 above)
Comment (Chef-Yogi) [reply apps/web/next.config.mjs:4]: I guess this is about the node version

File: apps/web/src/state/types.ts (hunk 6 above)
Comment (Chef-Yogi) [reply apps/web/src/state/types.ts:17]: @thechefpenguin this one update from the commits 2 weeks ago from you, could you help to take a look?

File: apps/web/next.config.mjs (hunk 3 above)
Comment (thechefpenguin) [reply apps/web/next.config.mjs:4]: @chefjackson I get an eslint error when using `with`. 
> Parsing error: ';' expected.  

That perhaps for static json imports we should use `assert`.  Not a major issue so I can revert back to `with` if we want

File: apps/web/src/components/CurrencyInputPanelSimplify/index.tsx (hunk 8 above)
Comment (Chef-Yogi) [reply apps/web/src/components/CurrencyInputPanelSimplify/index.tsx]: done

File: apps/web/src/state/types.ts (hunk 6 above)
Comment (thechefpenguin) [reply apps/web/src/state/types.ts:17]: @chefjackson This is coming from the values in design file. It is only for BSC.

File: apps/web/public/images/swap-spinner.png
[the remark this answers is not in the document]
Comment (Chef-Yogi) [reply apps/web/public/images/swap-spinner.png:1]: done

File: apps/web/next.config.mjs (hunk 3 above)
Comment (chefjackson) [reply apps/web/next.config.mjs:4]: let's revert back. I think we changed that due to higher nodejs version updated this convention

File: apps/web/public/images/swap-spinner.png
Comment (chefjackson) [inline apps/web/public/images/swap-spinner.png:1]: Please remove if we already added in assets

**Review by chefjackson — APPROVED** [review]

**Review by chef-ryan — APPROVED** [review]

**Pull request merged**